parasolr/query.py

```diff
@@ -158,6 +178,30 @@ def filter(self, *args, **kwargs) -> 'SolrQuerySet':
 
         return qs_copy
 
+    def facet(self, *args: str, **kwargs) -> 'SolrQuerySet':
+        """
+        Return a new SolrQuerySet with Solr faceting enabled and facet.field
+        parameter set. Does not yet support ranged faceting.
+
+        Subsequent calls will reset the facet.field to the last set of
+        args in the chain.
+
+        For example::
+
+            qs = queryset.facet('person_type', 'age')
+            qs = qs.facet('item_type')
+
+        would result in `item_type` being the only facet field.
+        """
+        qs_copy = self._clone()
+
+        # enable faceting
+        qs_copy.facet_opts['facet'] = True
+        # add args as a list to facet.field
+        qs_copy.facet_opts['facet.field'] = list(args)
+
```

Comment:
What about other facet options? Sort, min count. Could handle the way we're doing additional highlighting options, just use `kwargs`...

I'm less certain how to handle field-specific facet options, but maybe we hold off on that until we have a specific use case.

Reply:
If we want field-specific facets, _that_ might be a method like `facet_field` where you specify the field and then the field-specific facet options. (I don't think we need this right away, but does confirm my opinion to change the existing `facet_fields` method to `get_facets` or similar.)

Reply:
I had somewhat deliberately held off on the facet kwargs because I wasn't sure if we wanted to add complexity beyond what was needed for the story that we're making these adjustments for. However, just doing an update for kwargs would allow for using them without major issues, so I'll add that.